### PR TITLE
feat/LIVE-5657 DebugErrors tool and attempt at unified rendering

### PR DESCRIPTION
**Pull request by juan-cortes**
### 📝 Description
Part of a wider effort to align CS, product, design and engineering in the way we handle error cases in our flows. Allowing even non-tech savvy colleagues to check the rendering and suggest modifications to specific error cases. The tool can be accessed via the Debug menu as always and will prompt both fullscreen and modal versions of the selected errors.

### Potentially breaking changes
Before this gets merged we need to go over all the flows and screen that are impacted to be sure we are not blocking the user and still offer a way out of errors. In particular I am aware of breaking analytics in some flows but no fix makes sense until we know from a product perspective exactly what makes sense to track and whether this is the right place to do so.

### ❓ Context

- **Impacted projects**: `ledger-live-mobile` 
- **Linked resource(s)**: `https://ledgerhq.atlassian.net/browse/LIVE-5657` 

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 📸 Demo

https://user-images.githubusercontent.com/4631227/219034269-1a81fe89-a754-4efa-959a-a23a15762239.mov

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-mobile/src/components/CustomImageDeviceAction.tsx [hunk 1]
```diff
@@ -116,50 +103,21 @@ const CustomImageDeviceAction: React.FC<Props & { remountMe: () => void }> = ({
...
-                  : analyticsErrorTryAgainEventProps
-              }
-            >
-              {isRefusedOnStaxError
-                ? t("customImage.uploadAnotherImage")
-                : t("common.retry")}
-            </Button>
-            {isRefusedOnStaxError ? (
-              <Flex py={7}>
-                <Link
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/components/CustomImageDeviceAction.tsx:152]: just writing this here so we don't forget to put back a "skip" button somehow once the discussion with product/design has advanced

Comment (juan-cortes) [reply apps/ledger-live-mobile/src/components/CustomImageDeviceAction.tsx:152]: Likely we could have a conditional secondary action that is shown only when it receives the prop (like we discussed), it should be easy to implement and would be compatible with a device action version that doesn't have it.

File: apps/ledger-live-mobile/src/hooks/useErrorRenderData.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      break;
+
+    case error instanceof ManagerDeviceLockedError:
+      data.Icon = Icons.LockAltMedium;
+      data.iconColor = "warning.c80";
+      data.hasExportLogs = false;
+      break;
+
+    default:
+      // Yaiza
```
Comment (alexandremgo) [inline apps/ledger-live-mobile/src/hooks/useErrorRenderData.ts:125]: 😂

Comment (juan-cortes) [reply apps/ledger-live-mobile/src/hooks/useErrorRenderData.ts:125]: TS wanted something there, again, this PR is not for review really, it should've been a draft :trollface:

File: apps/ledger-live-mobile/e2e/specs/errors.spec.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        await debugSettingsPage.selectErrorByName(name);
+
+        if (type === "modal") {
+          // We want the modal view, so toggle it before the screenshot
+          await debugSettingsPage.toggleErrorRenderingType();
+        }
+
+        // We are in the tool now, create some sort of loop that goes over the errors and outputs
+        // screenshots of the device.
+        const dir = `/Users/juan.cortes-ext/Desktop/Screenshots/${type}/`;
```
Comment (alexandremgo) [inline apps/ledger-live-mobile/e2e/specs/errors.spec.ts:58]: This won't work on other computers - is it the way to save screenshots ?

File: apps/ledger-live-mobile/e2e/specs/errors.spec.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+beforeAll(async () => {
+  await loadConfig("1AccountBTC1AccountETH", true);
+  portfolioPage = new PortfolioPage();
+  settingsPage = new SettingsPage();
+  debugSettingsPage = new DebugSettingsPage();
+});
+
+describe("Error rendering tool", () => {
+  it("waits for Onboarding page to be ready", async () => {
```
Comment (alexandremgo) [inline apps/ledger-live-mobile/e2e/specs/errors.spec.ts:38]: Not sure it's what the test is doing 🙃

File: apps/ledger-live-mobile/e2e/specs/errors.spec.ts (hunk 3 above)
Comment (juan-cortes) [reply apps/ledger-live-mobile/e2e/specs/errors.spec.ts:58]: Good catch, I forgot this.

File: apps/ledger-live-mobile/e2e/specs/errors.spec.ts (hunk 4 above)
Comment (juan-cortes) [reply apps/ledger-live-mobile/e2e/specs/errors.spec.ts:38]: hahahaha copy paste me

**Comment by juan-cortes** [comment]
Rebased on develop, and addressed the issues @alexandremgo pointed out.
Still not mergable though.

**Comment by juan-cortes** [comment]
Let us close this PR for now, as @alexandremgo will attempt to retrieve any recoverable work the branch. We can reassess the situation at a later time, after our Product, Customer Service, and Design teams have agreed on the most appropriate way to display errors, including relevant support links, tracked events, calls-to-action, and wording. 
Once we have established these guidelines, we can proceed with implementing the necessary changes.

**Pull request closed without merging**